src/plugins/gopts/gopts.c

```diff
@@ -34,6 +34,35 @@ static void cleanupEnvp (char ** envp);
 #error "No implementation available"
 #endif
 
+/**
+ * Detects whether we are in help mode or not.
+ * DOES NOT set 'proc/elektra/gopts/help' for use with elektraGetOptsHelpMessage().
+ *
+ * @retval 1 if the -h or --help is part of argv
+ * @retval 0 otherwise
+ * @retval -1 on error (could not load argv)
+ */
+int elektraGOptsIsHelpMode (void)
+{
+	char ** argv = NULL;
+	int argc = loadArgs (&argv);
+
+	if (argv == NULL)
+	{
+		return -1;
+	}
+
+	for (int i = 0; i < argc; ++i)
+	{
+		if (strcmp (argv[i], "-h") == 0 || strcmp (argv[i], "--help") == 0)
```

Comment:
I do not like that this is hardcoded here. `-H` is very common (and also used in the `kdb` tool). `-h` has man other meanings.

Furthermore this is not properly parsed, e.g. `-vh` would not be recognized.

Reply:
> I do not like that this is hardcoded here. `-H` is very common (and also used in the `kdb` tool). `-h` has man other meanings.

Using `-h`/`--help` for help mode is already hard-coded in `elektraGetOpts` (there it is properly parsed).

This function is not actually intended to accurately detect help mode (for that one should still check for the presence of `proc/elektra/gopts/help`. It is only meant in case `kdbGet()` failed.

Apart from `kdb` tool, I have never seen `-H` used in favour of `-h`. The `kdb` tool also takes a very unusual approach, since it opens the man-page instead of printing to stdout/stderr.
In fact most tools I know, use only `--help` (if they support long options).

> Furthermore this is not properly parsed, e.g. `-vh` would not be recognized.

See above https://github.com/ElektraInitiative/libelektra/pull/2944#issuecomment-531577466.

Reply:
> Apart from kdb tool, I have never seen -H used in favour of -h.

Because of how the cmdline options in Elektra are implemented every option needs to have a short option. `-H` was the most natural one as `-h` is already "human readable".

> The kdb tool also takes a very unusual approach, since it opens the man-page instead of printing to stdout/stderr.

It makes a lot of sense to not have a --help output and a separated man page. E.g. git also does it this way.  

> In fact most tools I know, use only --help (if they support long options).

I would prefer if we only hardcode `--help` and drop support for `-h`.